controllers/vspherecluster_controller.go

```diff
@@ -181,18 +180,6 @@ func (r clusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr er
 	// Always issue a patch when exiting this function so changes to the
 	// resource are patched back to the API server.
 	defer func() {
-		// always update the readyCondition.
```

Comment:
Don't think moving this into `func Patch()` changed the functionality but curious about the reason for this change. It felt more transparent to have the spec/status changes located in the controller and have patch just attempt to patch the object in the current state.

Reply:
Sure!
When a condition changes, we are required to update the summary condition as well, otherwise `ready` gets out of sync from the underlying conditions.
Given that in many controllers, we are starting to issue `Patch` in the middle of the reconciliation loop, I'm systematically moving "update the readyCondition" into `func Patch()` to ensure that we are respecting the requirement above.

Reply:
That makes sense. Thanks!